modules/sync/unique_queue.go

```diff
@@ -29,11 +28,43 @@ func NewUniqueQueue(queueLength int) *UniqueQueue {
 	}
 
 	return &UniqueQueue{
-		table: NewStatusTable(),
-		queue: make(chan string, queueLength),
+		table:  NewStatusTable(),
+		queue:  make(chan string, queueLength),
+		closed: make(chan struct{}),
+	}
+}
+
+// Close closes this queue
+func (q *UniqueQueue) Close() {
+	select {
+	case <-q.closed:
+	default:
+		q.table.lock.Lock()
+		select {
+		case <-q.closed:
+		default:
+			close(q.closed)
+		}
+		q.table.lock.Unlock()
 	}
 }
 
+// IsClosed returns a channel that is closed when this Queue is closed
+func (q *UniqueQueue) IsClosed() <-chan struct{} {
+	return q.closed
+}
+
+// IDs returns the current ids in the pool
+func (q *UniqueQueue) IDs() []interface{} {
```

Comment:
Where is this needed?

Reply:
It's not needed in this PR

Reply:
If it returns a snapshot of the IDs of an ever-changing queue (instead of draining it), I'd suggest renaming it to `GetIDsSnapshot()` or something. Otherwise, when it's used for another PR its functionality will probably be overlooked and not reviewed properly.

Reply:
Heh, it's to be used once the queue is closed... Patience my friend..